File: java_projects/glue_code/commons-pool/src/main/java/org/apache/commons/pool2/impl/NoOpCallStack.java
package org.apache.commons.pool;

import java.io.PrintWriter;
import org.graalvm.polyglot.Value;

public class NoOpCallStack implements CallStack {
  public static final CallStack INSTANCE = new NoOpCallStack();
  private static Value clz =
      ContextInitializer.getPythonClass("/NoOpCallStack.py", "NoOpCallStack");
  private Value obj;

  public NoOpCallStack(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public boolean printStackTrace(final PrintWriter writer) {
    //
    // return false;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("printStackTrace", writer).as(boolean.class);
  }

  public void fillInStackTrace() {
    //

    obj.invokeMember("fillInStackTrace");
  }

  public void clear() {
    //

    obj.invokeMember("clear");
  }

  private NoOpCallStack() {
    //

    this.obj = clz.invokeMember("__init__");
  }
}

File: data/skeletons/commons-pool/src/main/org/apache/commons/pool2/impl/NoOpCallStack.py
# Imports Begin
from src.main.org.apache.commons.pool2.impl.CallStack import *
import io

# Imports End


class NoOpCallStack(CallStack):

    # Class Fields Begin
    INSTANCE: CallStack = None
    # Class Fields End

    # Class Methods Begin
    def printStackTrace(
        self, writer: typing.Union[io.TextIOWrapper, io.StringIO]
    ) -> bool:
        pass

    def fillInStackTrace(self) -> None:
        pass

    def clear(self) -> None:
        pass

    def __init__(self) -> None:
        pass

    # Class Methods End
